Maximum node count documentation (#850)
astronomer/docs: astro/resource-reference-aws.md

> | [Route Tables](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Route_Tables.html) | Home for the routes. | 2x |

Reviewer: Can't comment but suggestion for the `EC2 Instances` box:

> EC2 instances (nodes) power all system and Airflow components on Astro, including workers and schedulers. EC2 instances auto-scale based on the demand for nodes in your Cluster.

| [VPC](https://docs.aws.amazon.com/vpc/latest/userguide/what-is-amazon-vpc.html) | Virtual network for launching and hosting AWS resources. | 1x /19 |
| [S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide//Welcome.html) | S3 bucket for storage of Airflow task logs. | 1x |

> | Max Node Count | The maximum number of EC2 worker nodes that your Astro Cluster can support. When this limit is reached, your Astro Cluster can't auto-scale and worker pods may fail to schedule. | 20 |

Author: Is it only worker Pods that will fail to schedule? Or will there be other errors as well?

Decided not to capitalize "node" here because K8s is so inconsistent with the capitalization that we should just decide for ourselves, and there's no other "node" in our system so I think people will know what we mean regardless of whether we capitalize it.

> | Max Node Count | The maximum number of EC2 worker nodes that your Astro Cluster can support at any given time. Once this limit is reached, your Cluster cannot auto-scale and worker pods may fail to schedule. | 20 |

Reviewer 2: ```suggestion
| Max Node Count | The maximum number of EC2 worker nodes that your Astro Cluster can support. When this limit is reached, your Astro Cluster can't auto-scale and worker pods may fail to schedule. | 20 |
```

| [Route Tables](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Route_Tables.html) | Home for the routes. | 2x |
| [VPC](https://docs.aws.amazon.com/vpc/latest/userguide/what-is-amazon-vpc.html) | Virtual network for launching and hosting AWS resources. | 1x /19 |
| [S3](https://docs.aws.amazon.com/AmazonS3/latest/userguide//Welcome.html) | S3 bucket for storage of Airflow task logs. | 1x |

> | Max Node Count | The maximum number of EC2 nodes that your Astro Cluster can support. When this limit is reached, your Astro Cluster can't auto-scale and worker Pods may fail to schedule. | 20 |

Reviewer: ```suggestion
| Maximum Node Count | The maximum number of EC2 nodes that your Astro cluster can support. When this limit is reached, your Astro cluster can't auto-scale and worker Pods may fail to schedule. | 20 |
```

Sorry, forgot the lowercase `cluster` thing 🤦‍♀️

## Supported Cluster Configurations

You might need to modify configurations of a new or existing cluster on Astro. This section provides a reference for cluster configuration options.

> To create a new Cluster on Astro with a specified configuration, see [Create a Cluster](create-cluster.md). To request a change to an existing Cluster, see [Modify a Cluster](modify-cluster.md). Astronomer is currently responsible for completing all cluster configuration changes.

Reviewer: ```suggestion
To create a new cluster on Astro with a specified configuration, see [Create a Cluster](create-cluster.md). To request a change to an existing cluster, see [Modify a Cluster](modify-cluster.md). Astronomer is currently responsible for completing all cluster configuration changes.
```

:::

### Maximum node count

Each Astro cluster has a maximum node count, which is the maximum number of worker nodes that can run at any given time on a cluster. This limit does not apply to other nodes such as RDS instances or Astronomer-managed node pools.

> The default maximum node count for a cluster is 20. This value can be changed for any cluster by Astronomer.

Author: Updated this suggestion slightly, as both the original wording and the suggestion give the impression that Astronomer has the autonomy to change this number when they see fit without user input.

> The default maximum node count for a cluster is 20. When required, you can change this value by contacting [Astronomer support](https://support.astronomer.io).
> 
> The number of worker nodes in your cluster [autoscales](configure-deployment-resources.md#worker-autoscaling-logic) based on how many tasks are currently queued or running. If a cluster's worker node count scales to the maximum node count, new tasks can't run or be scheduled.

Reviewer: ```suggestion
The default maximum node count for a cluster is 20. This maximum applies to all nodes in your cluster. As defined in [Kubernetes documentation](https://kubernetes.io/docs/tutorials/kubernetes-basics/explore/explore-intro/), multiple Pods can run on a single Node depending on the size of each.

Typically, worker Pods increase or decrease in quantity more often than other types of Pods. The size and number of worker Pods directly affect the total of number of Nodes in your cluster. Worker Pods autoscale per Deployment based on the number of tasks that are queued or running. See [Worker autoscaling logic](configure-deployment-resources.md#worker-autoscaling-logic). When you configure **Worker Resources** to determine the size of workers for a Deployment, your cluster's total node count may increase or decrease. 

If the node count for your Cluster reaches maximum node count, new tasks may not run or get scheduled. Astronomer monitors maximum node count and is responsible for contacting your organization if it is reached. To check your cluster's node count at any time, contact [Astronomer Support](https://support.astronomer.io).
```

Author: Getting confused about when to use node vs. pod here. I also think that we are restating multiple times across this text that "nodes can scale, and worker nodes scale the most", so I tried to cut this suggestion down to that essence.

Reaching the maximum node count results in the following error in a Deployment's [scheduler logs](view-logs.md):

```text
│   Warning  FailedScheduling  6s (x10 over 11m)  default-scheduler  0/6 nodes are available: 6 Insufficient cpu.                                                                                                                                                                                                    │

> │ I1110 19:41:16.548235       1 static_autoscaler.go:392] Max total nodes in cluster reached       

Reviewer 3: Where did this info come from?

Author: @stuart23 I got it from the [original issue](https://github.com/astronomer/docs/issues/845), and I see some similar errors in Slack.

Reviewer 3: @paolaperaza I see you added it to the issue? Where did this come from?

I ask because these are kubernetes events, and would not show up in the schedulers log stream at all

```

> If you see this error in a Deployment, reach out to [Astronomer Support](https://support.astronomer.io) and request an increase to your cluster's maximum node count.

Reviewer 2: ```suggestion
If you see this error in a Deployment, contact [Astronomer Support](https://support.astronomer.io) and request an increase of your cluster's maximum node count.
```

> Reaching the maximum node count results in the following error in a Deployment's [scheduler logs](view-logs.md):
> 
> ```text
> │   Warning  FailedScheduling  6s (x10 over 11m)  default-scheduler  0/6 nodes are available: 6 Insufficient cpu.                                                                                                                                                                                                    │
> │ I1110 19:41:16.548235       1 static_autoscaler.go:392] Max total nodes in cluster reached       
> ```
> 
> If you see this error in a Deployment, contact [Astronomer Support](https://support.astronomer.io) and request an increase of your cluster's maximum node count.
> 

Reviewer: ```suggestion
```

I was wrong, @jwitz -- sorry 😞 Let's just delete this.

| t3.xlarge          | 27         | 2.7 CPUs  | 10.1 GiB MEM |
| t3.2xlarge         | 67         | 6.7 CPUs  | 25.1 GiB MEM |

> The maximum value for **Worker Resources** on Astro is 400 AU, which means that an Astro cluster might not make full use of the CPU and memory capacity of some node instance types. In the table, these node instance types are marked with an asterisk. If your Organization is interested in using an instance type that supports a worker size limit higher than 400 AU, contact [Astronomer Support](https://support.astronomer.io). For more information about configuring worker size on Astro, see [Configure a Deployment](configure-deployment-resources.md#worker-resources).

Author: ```suggestion
Because the maximum value for **Worker Resources** is 400 AU, a cluster might not be able to use all of the available CPU and memory capacity for some node instance types. In the table, these node instance types are marked with an asterisk. If your Organization is interested in using an instance type that supports a worker size limit higher than 400 AU, contact [Astronomer Support](https://support.astronomer.io). For more information about configuring worker size on Astro, see [Configure a Deployment](configure-deployment-resources.md#worker-resources).
```
